DependencyInjection/DoctrineExtension.php

```diff
@@ -842,11 +842,17 @@ private function loadMessengerServices(ContainerBuilder $container) : void
             $container->removeDefinition('doctrine.orm.messenger.event_subscriber.doctrine_clear_entity_manager');
         }
 
+        $transportFactoryDefinition = $container->getDefinition('messenger.transport.doctrine.factory');
         if (! class_exists(DoctrineTransportFactory::class)) {
-            return;
+            // If symfony/messenger < 5.1
+            if (! class_exists(\Symfony\Component\Messenger\Transport\Doctrine\DoctrineTransportFactory::class)) {
```

Comment:
You could use a `use` statement with an alias, like `LegacyDoctrineTransportFactory`. Makes this look just a little nicer ;)

Reply:
@alcaeus I'm not sure, but what about use statements to nonexisting classes?

In this situation, I would prefer to drop the use statements and use the FQCN in code. 🤔

Reply:
I slightly prefer @alcaeus suggestion just because it's used on two lines, but this is really petty discussion, so let's not block merge because of it.

Reply:
Yep, this can always be fixed later.